packages/nx/src/command-line/migrate.ts

```diff
@@ -1040,7 +1040,12 @@ async function runNxMigration(root: string, packageName: string, name: string) {
 
   const collection = readJsonFile<MigrationsJson>(collectionPath);
   const g = collection.generators || collection.schematics;
-  const implRelativePath = g[name].implementation || g[name].factory;
+  const implRelativePath = g[name]?.implementation || g[name]?.factory;
+  if (!implRelativePath) {
+    throw new Error(
+      `Can not get relative path for migration's implementation as collection not found by name="${name}" for package="${packageName}", collectionPath="${collectionPath}"`
```

Comment:
This messaging is still not quite accurate. The collection was found, but the implementation was not. I'd recommend something like this:
```suggestion
      `Unable to determine implementation path for ${collectionPath}:${name}`
```

The `collection:generator` pattern matches expectations from `nx g ...`, so I think its more readable.

Reply:
Have a look at new message? I've added also indication if `generators ` or `schematics` was used.